### PR TITLE
Ant task: do not include runtime by default if destination is a jar

### PR DESCRIPTION
**Pull request by scaventz**
[KT-16227](https://youtrack.jetbrains.com/issue/KT-16227)

### CODE REVIEW

File: ant/src/org/jetbrains/kotlin/ant/Kotlin2JvmTask.kt [hunk 1]
```diff
@@ -71,6 +71,6 @@ class Kotlin2JvmTask : KotlinCompilerBaseTask() {
 
         if (noStdlib) args.add("-no-stdlib")
         if (noReflect) args.add("-no-reflect")
-        if (includeRuntime) args.add("-include-runtime")
+        if (includeRuntime && output?.extension == "jar") args.add("-include-runtime")
```
Comment (udalov) [inline ant/src/org/jetbrains/kotlin/ant/Kotlin2JvmTask.kt:74]: Could you explain this change?

I think that in the case of an incorrect config, such as `includeRuntime` enabled + directory output, it's better to still pass everything to the compiler as specified by the user, in order for the compiler to handle it in an expected way, such as for example report a warning, than to silently change the compiler arguments here.

With that said, the compiler right now does _not_ report any warning if `-include-runtime` is passed with a directory output, but that is a separate issue (it probably should).

File: compiler/testData/integration/ant/jvm/kt16227/build.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<project name="Ant Task Test" default="build">
+    <taskdef resource="org/jetbrains/kotlin/ant/antlib.xml" classpath="${kotlin.lib}/kotlin-ant.jar"/>
+    <target name="build">
+        <kotlinc src="${test.data}" output="${temp}/test.jar">
+        </kotlinc>
+
+    </target>
+</project>
```
Comment (udalov) [inline compiler/testData/integration/ant/jvm/kt16227/build.xml:8]: Looks like this test doesn't check anything that isn't already checked by other tests.

I suppose we need a test that somehow checks that with the jar output, no `includeRuntime` leads to the jar file being created without stdlib classes in it.

And also, another test that does use the `includeRuntime` option of the Ant plugin (currently there are zero tests on it) and checks that the resulting jar file has the stdlib classes.

**Review by udalov — CHANGES_REQUESTED** [review]

File: ant/src/org/jetbrains/kotlin/ant/Kotlin2JvmTask.kt (hunk 1 above)
Comment (scaventz) [reply ant/src/org/jetbrains/kotlin/ant/Kotlin2JvmTask.kt]: I thought that it doesn't make sense passing ```-include-runtime``` to compiler when destination is a jar, obviously I was wrong. It seems the only thing we need to do is to change the default value of  ```includeRuntime``` from ```true``` to ```false```.

File: compiler/testData/integration/ant/jvm/kt16227/build.xml (hunk 2 above)
Comment (scaventz) [reply compiler/testData/integration/ant/jvm/kt16227/build.xml]: A new test has been added, which considered three different situations(includeRuntime=true/false/not specified).

**Review by udalov — APPROVED** [review]

**Comment by udalov** [comment]
Merged with a minor addition to the test.

Thank you for the contribution! :)
